src/siemens-s7_fmt_plug.c

```diff
@@ -93,6 +93,10 @@ static int valid(char *ciphertext, struct fmt_main *self)
 		goto bail;
 	if (strlen(p) != 40)
 		goto bail;
+	if ((p = strtok(NULL, "$")) == NULL)	/* Fix bug: #1090 */
+		goto bail;
+	if (strlen(p) < 40)
+		goto bail;
```

Comment:
I think @kholia is the author. But surely it should be exactly 40 - we are later reading the fixed BINARY_SIZE from it.